Makd.mak

```diff
@@ -565,6 +565,40 @@ $O/test-%.stamp: $O/test-%
 	$(call exec,$< $(ITFLAGS),$<,run)
 	$Vtouch $@
 
+# examples rules
+##########################
+
+# examples are assumed to be standalone programs, so we recursively search
+# for files examples/*.d and assume that each file is a separate example
+# program. Files can exist in sub-directories under examples/ dir.
+# The sources list is filtered through the $(EXAMPLES_FILTER_OUT) variable 
+# contents (using the Make function $(filter-out)), so you can exclude an
+# example by adding the example file path (as an absolute path using $C) to
+# this variable.
+
+# The 'example' target builds all the examples.
+.PHONY: example
+example: $(patsubst $T/example/%.d,$O/example/%,\
+	$(filter-out $(EXAMPLE_FILTER_OUT),$(shell find $T/example/ -name "*.d")))
```

Comment:
> 2. More complex examples that need multiple files, that should be in <name>/main.d

I was thinking about this, however I was thinking that maybe a directory will map several related sub-system files together. And my thinking was if one file is deemed to be imported and not stand-alone compiled then you'd add it to the `EXAMPLE_FILTER_OUT` example.

The best case I think would to check if the module has a main function, if so then we compile it, if not then it's being imported. Do you know if we can do this?